pyomo/contrib/parmest/utils/model_utils.py

```diff
@@ -124,14 +128,14 @@ def convert_params_to_vars(model, param_names=None, fix_vars=False):
     if len(substitution_map) == 0:
         return model
 
-    # Update the list of param_names if the parameters were indexed
-    if len(indexed_param_names) > 0:
-        param_names = indexed_param_names
+    # Update the list of param_CUIDs if the parameters were indexed
+    if len(indexed_param_CUIDs) > 0:
+        param_CUIDs = indexed_param_CUIDs
 
     # Convert Params to Vars in Expressions
     for expr in model.component_data_objects(pyo.Expression):
         if expr.active and any(
-            v.name in param_names for v in identify_mutable_parameters(expr)
+            ComponentUID(v) in param_CUIDs for v in identify_mutable_parameters(expr)
```

Comment:
This is *very* inefficient (quadratic in the number of params).  If you are going to test membership, `param_CUIDs` needs to be a set / dict / mapping object (so that lookups are O(1) and not O(n)).

Reply:
Converted list to set for the lookups at the end of the function.